src/Algebra/Apartness/Structures.agda

```diff
@@ -0,0 +1,58 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Algebraic structures with an apartness relation
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Algebra.Apartness.Structures where
+
+open import Algebra.Core using (Op₁; Op₂)
+open import Relation.Binary.Core using (Rel)
+open import Level using (_⊔_; suc)
+open import Data.Product using (∃-syntax; _×_; _,_; proj₂)
+open import Algebra.Definitions using (Invertible)
+open import Algebra.Structures using (IsCommutativeRing)
+open import Relation.Binary.Structures using (IsEquivalence; IsApartnessRelation)
+open import Relation.Binary.Definitions using (Tight)
+open import Relation.Nullary using (¬_)
+import Relation.Binary.Properties.ApartnessRelation as ApartnessRelation
+
+private
+  Iso : ∀ {a b} → Set a → Set b → Set _
+  Iso A B = (A → B) × (B → A)
+
+record IsHeytingCommutativeRing
+  {c ℓ₁ ℓ₂} {Carrier : Set c}
+  (_≈_ : Rel Carrier ℓ₁)
+  (_#_ : Rel Carrier ℓ₂)
```

Comment:
I think you went too far the other way again. Every parameter above this comment will be common to _all_ structures defined in this file, and therefore should be top-level module parameters, whereas everything below it will vary depending on the structure in question and therefore should parameters to the individual records. My fault for not being clearer, apologies!

Reply:
If you do that then you can also import `Algebra.Definitions`, `Algebra.Structures` and `Relation.Binary.Structures` parameterised by the equality relation which will cut down a bit on the boilerplate!